src/services/rfqm_service.ts

```diff
@@ -432,11 +436,90 @@ export class RfqmService {
     /**
      * Process an orderHash as an RfqmJob. Throws an error if job must be retried
      */
-    // tslint:disable: prefer-function-over-method
     public async processRfqmJobAsync(orderHash: string): Promise<void> {
-        logger.info({ orderHash }, 'finished processing job');
-        // tslint:disable-next-line: custom-no-magic-numbers
-        return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
+        logger.info({ orderHash }, 'start processing job');
+
+        // Get job
+        const job = await this.dbUtils.findJobByOrderHashAsync(orderHash);
+        if (job === undefined) {
+            throw new NotFoundError(`job for orderHash ${orderHash} not found`);
+        }
+
+        // Basic validation
+        const { calldata, makerUri, order, fee } = job;
+        if (calldata === undefined) {
```

Comment:
can we throw these checks in another function, e.g. validateJob()?

Reply:
yeah, that's much better